Doc/library/code.rst

```diff
@@ -23,27 +23,33 @@ build applications which provide an interactive interpreter prompt.
    ``'__doc__'`` set to ``None``.
 
 
-.. class:: InteractiveConsole(locals=None, filename="<console>")
+.. class:: InteractiveConsole(locals=None, filename="<console>", block_exit=False)
 
    Closely emulate the behavior of the interactive Python interpreter. This class
    builds on :class:`InteractiveInterpreter` and adds prompting using the familiar
-   ``sys.ps1`` and ``sys.ps2``, and input buffering.
+   ``sys.ps1`` and ``sys.ps2``, and input buffering. If *block_exit* is True,
+   ``exit()`` and ``quit()`` will not terminate the whole process, instead they
+   will exit from the interactive console.
 
+   .. versionchanged:: 3.12
+      Added *block_exit* parameter.
 
-.. function:: interact(banner=None, readfunc=None, local=None, exitmsg=None)
+.. function:: interact(banner=None, readfunc=None, local=None, exitmsg=None, block_exit=False)
 
    Convenience function to run a read-eval-print loop.  This creates a new
    instance of :class:`InteractiveConsole` and sets *readfunc* to be used as
-   the :meth:`InteractiveConsole.raw_input` method, if provided.  If *local* is
-   provided, it is passed to the :class:`InteractiveConsole` constructor for
-   use as the default namespace for the interpreter loop.  The :meth:`interact`
-   method of the instance is then run with *banner* and *exitmsg* passed as the
-   banner and exit message to use, if provided.  The console object is discarded
-   after use.
+   the :meth:`InteractiveConsole.raw_input` method, if provided.  If *local*
+   or *block_exit* is provided, it is passed to the :class:`InteractiveConsole`
+   constructor for use as the default namespace for the interpreter loop.
+   The :meth:`interact` method of the instance is then run with *banner* and
+   *exitmsg* passed as the banner and exit message to use, if provided.
+   The console object is discarded after use.
```

Comment:
We don't normally reflow the paragraph, it distracts the review. Just let the lines be uneven in length. :-)

Reply:
Ah, I just thought the paragraph would be kind of ugly. Do you want me to convert this back to the uneven format? So only one line is modified?

Reply:
Yes please! :)

Reply:
This still needs to be done.